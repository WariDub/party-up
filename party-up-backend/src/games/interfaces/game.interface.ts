export interface Game {
    id: number;
    cover: { [key: string]: string };
    genres: { [key: string]: string };
    name: string;
    rating: number;
}
