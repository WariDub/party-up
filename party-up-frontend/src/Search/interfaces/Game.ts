export interface Game {
  id: string;
  cover: GameCover;
  genres: { [key: string]: string };
  name: string;
  rating: number;
}

interface GameCover {
  url: string;
}
