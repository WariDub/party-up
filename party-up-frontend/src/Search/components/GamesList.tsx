import { Box, GridList, GridListTile, GridListTileBar } from '@material-ui/core';
import React from 'react';
import { Game } from '../interfaces/Game';

export interface GamesListProps {
  results: Game[];
}

export interface GamesListState {}

const GamesList = class extends React.PureComponent<GamesListProps, GamesListState> {
  render(): JSX.Element {
    const { results } = this.props;

    return (
      <Box p={1} mx="auto" width="75%">
        <GridList spacing={8} cols={3}>
          {results.map((result) => (
            <GridListTile key={result.id} cols={1}>
              <img src={result.cover.url.replace('t_thumb', 't_720p')} alt={result.name} />
              <GridListTileBar title={result.name} />
            </GridListTile>
          ))}
        </GridList>
      </Box>
    );
  }
};

export default GamesList;
