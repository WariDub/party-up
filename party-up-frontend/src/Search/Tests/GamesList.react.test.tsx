import  GamesList  from '../components/GamesList';
import { shallow, mount } from 'enzyme';
import { createBrowserHistory } from 'history';
import { matchPath } from 'react-router';
import { Game } from '../interfaces/Game';
import renderer from 'react-test-renderer';

describe(GamesList, () =>{
    test('It should mount', () => {
        let history = createBrowserHistory();
        history.push('/');

        let location = history.location;

        let game: Game = {
            id: 'TestId',
            name: 'TestName'
        }

        let match = matchPath('/ProfilePage', {
            path: '/ProfilePage',
            exact: true,
            strict: false,
        });

        let results: Game[] = [game];

        let component = shallow(<GamesList history = { history } location = { location } match = { match }/>);

        expect(component.length).toBe(1);
    });

    test('It should mount', () => {
        let history = createBrowserHistory();
        history.push('/');

        let location = history.location;

        let game: Game = {
            id: 'TestId',
            name: 'TestName'
        }

        let match = matchPath('/ProfilePage', {
            path: '/ProfilePage',
            exact: true,
            strict: false,
        });

        let results: Game[] = [game];

        let component = renderer.create(<GamesList history = { history } location = { location } match = { match }/>).toJSON();

        expect(component).toMatchSnapshot();
    });
});
