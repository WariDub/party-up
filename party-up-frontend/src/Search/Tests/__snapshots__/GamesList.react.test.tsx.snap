// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesList It should mount 1`] = `
<div
  className="MuiBox-root MuiBox-root-1"
>
  <ul
    className="MuiGridList-root"
    style={
      Object {
        "margin": -4,
      }
    }
  />
</div>
`;
